README.md

```diff
@@ -110,89 +115,140 @@ trainer.train()
 
 ### `RewardTrainer`
 
-This is a basic example of how to use the `RewardTrainer` from the library. The `RewardTrainer` is a wrapper around the `transformers` Trainer to easily fine-tune reward models or adapters on a custom preference dataset.
+Here is a basic example on how to use the `RewardTrainer`:
 
 ```python
 # imports
+from trl import RewardConfig, RewardTrainer
+from trl.extras.dataset_formatting import conversations_formatting_function
+from datasets import load_dataset
 from transformers import AutoModelForSequenceClassification, AutoTokenizer
-from trl import RewardTrainer
 
-# load model and dataset - dataset needs to be in a specific format
-model = AutoModelForSequenceClassification.from_pretrained("gpt2", num_labels=1)
-tokenizer = AutoTokenizer.from_pretrained("gpt2")
-
-...
+# load model and tokenizer
+tokenizer = AutoTokenizer.from_pretrained("Qwen/Qwen2.5-0.5B-Instruct")
+model = AutoModelForSequenceClassification.from_pretrained(
+    "Qwen/Qwen2.5-0.5B-Instruct", num_labels=1
+)
+model.config.pad_token_id = tokenizer.pad_token_id
+
+# load dataset and preprocess
+dataset = load_dataset("trl-lib/Capybara-Preferences", split="train")
+
+def preprocess_function(examples):
+    new_examples = {
+        "input_ids_chosen": [],
+        "attention_mask_chosen": [],
+        "input_ids_rejected": [],
+        "attention_mask_rejected": [],
+    }
+    for chosen, rejected in zip(examples["chosen"], examples["rejected"]):
+        tokenized_chosen = tokenizer(chosen)
+        tokenized_rejected = tokenizer(rejected)
+        new_examples["input_ids_chosen"].append(tokenized_chosen["input_ids"])
+        new_examples["attention_mask_chosen"].append(tokenized_chosen["attention_mask"])
+        new_examples["input_ids_rejected"].append(tokenized_rejected["input_ids"])
+        new_examples["attention_mask_rejected"].append(
+            tokenized_rejected["attention_mask"]
+        )
+
+    return new_examples
+
+chosen_fn = conversations_formatting_function(tokenizer, "chosen")
+rejected_fn = conversations_formatting_function(tokenizer, "rejected")
+dataset = dataset.map(lambda x: {"chosen": chosen_fn(x), "rejected": rejected_fn(x)})
+dataset = dataset.map(
+    preprocess_function,
+    batched=True,
+)
 
-# load trainer
+# configure trainer
+args = RewardConfig(
+    per_device_train_batch_size=2,
+    remove_unused_columns=False,
+    output_dir="Qwen2.5-0.5B-Reward",
+)
 trainer = RewardTrainer(
     model=model,
     tokenizer=tokenizer,
     train_dataset=dataset,
+    args=args,
 )
 
 # train
 trainer.train()
 ```
 
-### `PPOTrainer`
+### `RLOOTrainer`
```

Comment:
I switched to RLOO because (a) it is better than PPO and (b) we're deprecating `PPOTrainer` and I didn't want to use `PPOv2Trainer` which looks a bit experimental in name